subsys/bluetooth/mesh/prov.c

```diff
@@ -107,10 +107,98 @@ static bt_mesh_input_action_t input_action(uint8_t action)
 	}
 }
 
-int bt_mesh_prov_auth(uint8_t method, uint8_t action, uint8_t size)
+static int check_output_auth(bt_mesh_output_action_t output, uint8_t size)
+{
+	if (!output) {
+		return -EINVAL;
+	}
+
+	if (!(bt_mesh_prov->output_actions & output)) {
+		return -EINVAL;
+	}
+
+	if (size > bt_mesh_prov->output_size) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int check_input_auth(bt_mesh_input_action_t input, uint8_t size)
+{
+	if (!input) {
+		return -EINVAL;
+	}
+
+	if (!(bt_mesh_prov->input_actions & input)) {
+		return -EINVAL;
+	}
+
+	if (size > bt_mesh_prov->input_size) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static void get_auth_string(unsigned char *str, uint8_t size)
+{
+	bt_rand(str, size);
+
+	/* Normalize to '0' .. '9' & 'A' .. 'Z' */
+	for (uint8_t i = 0U; i < size; i++) {
+		str[i] %= 36;
+		if (str[i] < 10) {
+			str[i] += '0';
+		} else {
+			str[i] += 'A' - 10;
+		}
+	}
```

Comment:
It is the existing algorithm that I moved in a separate function to not encumber huge auth switch case. Do you want me to refactor it here?

Reply:
Right, sorry. We could do this separately, it's not directly relevant to this PR then.

Reply:
Applied your changes since the current algorithm generates unreadable symbols if change type 'unsigned char' to 'char'. Otherwise, it requires unnecessary types convertions that @PavelVPV  suggested to remove.